src/lib/utils/ct_utils.h

```diff
@@ -107,6 +107,54 @@ constexpr inline T value_barrier(T x) {
    }
 }
 
+/**
+* A Choice is used for constant-time conditionals.
+*
+* Internally it always is either all 0 bits or all 1 bits.
+*/
+class Choice final {
+   public:
+      /**
+      * If v == 0 return an unset (false) Choice, otherwise a set Choice
+      */
+      template <typename T>
+      constexpr static Choice from_int(T v)
+         requires(std::is_unsigned<T>::value)
+      {
+         const auto v_is_0 = ct_is_zero<T>(value_barrier<T>(v));
+         return Choice(ct_is_zero<uint32_t>(static_cast<uint32_t>(v_is_0)));
+      }
+
+      template <typename T>
+      constexpr T choose(T a, T b)
+         requires(std::is_unsigned<T>::value)
+      {}
+
+      constexpr Choice operator!() const { return Choice(~value()); }
+
+      constexpr Choice operator&&(const Choice& other) const { return Choice(value() & other.value()); }
+
+      constexpr Choice operator||(const Choice& other) const { return Choice(value() | other.value()); }
+
+      constexpr Choice operator!=(const Choice& other) { return Choice(value() ^ other.value()); }
+
+      /// Unsafe conversion to bool
+      constexpr bool as_bool() const { return m_value != 0; }
+
+      /// Return the masked value
+      constexpr uint32_t value() const { return value_barrier(m_value); }
```

Comment:
If we ensure that all "constructors" use a `value_barrier`, we (technically) shouldn't need it for this accessor, as the value in `m_value` should never be predictable for the compiler. But then again: better safe than sorry, I guess.

Reply:
I would agree with you in theory!

There are I suppose two different ways we could approach this

1. Use what seems like it should be sufficient (eg, just in the constructor). Test const-timeness with latest available compilers. If something fails, we need [more `value_barrier`](https://www.youtube.com/watch?v=cVsQLlk-T0s)
2. Use it kind of wherever. (Current shotgun approach)

Part of the problem is we're trying to anticipate any compiler optimizations that might be made in the future. But I think the bigger issue is that our current const timeness testing is just insufficient to get a good signal from (1)